scripts/redux/actions.js

```diff
@@ -137,3 +137,110 @@ const teamActions = {
       });
   }
 };
+
+const userActions = {
+  signIn: () => {
+    let provider = new firebase.auth.GoogleAuthProvider();
+    return firebase.auth()
+      .signInWithPopup(provider)
+      .then((signInObject) => {
+
+        if (navigator.credentials) {
+          var cred = new FederatedCredential({
+            id: signInObject.user.email,
+            name: signInObject.user.displayName,
+            iconURL: signInObject.user.photoURL,
+            provider: 'https://accounts.google.com'
+          });
+
+          navigator.credentials.store(cred);
+        }
+
+        store.dispatch({
+          type: SIGN_IN,
+          user: Object.assign(signInObject.user, { signedIn: true })
```

Comment:
in this case you modify `signInObject.user`. To create a new one do:
```
Object.assign({}, signInObject.user, { signedIn: true})
```
or spread operator
```
{
 ...signInObject.user,
 signedIn: true
}
```